ios_sample_test_app/ios_sample_test_app/ViewController.swift

```diff
@@ -0,0 +1,488 @@
+        buo.title = "My Content Title"
+        buo.contentDescription = "My Content Description"
+        buo.imageUrl = "https://lorempixel.com/400/400"
+        buo.publiclyIndex = true
+        buo.locallyIndex = true
+        buo.contentMetadata.customMetadata["key1"] = "value1"
+        
+        let lp: BranchLinkProperties = BranchLinkProperties()
+        lp.channel = "facebook"
+        lp.feature = "sharing"
+        lp.campaign = "content 123 launch"
+        lp.stage = "new user"
+        lp.tags = ["one", "two", "three"]
+
+        lp.addControlParam("$desktop_url", withValue: "http://example.com/desktop")
+        lp.addControlParam("$ios_url", withValue: "http://example.com/ios")
+        
+        let qrCode = BranchQRCode()
+        qrCode.codeColor = UIColor.white
+        qrCode.backgroundColor = UIColor.blue
+        qrCode.centerLogo = "https://i.snipboard.io/nshJlY.jpg"
+        qrCode.width = 350
+        qrCode.margin = 0
+        qrCode.imageFormat = .JPEG
+        
+        qrCode.getAsImage(buo, linkProperties: lp) { qrCodeImage, error in
+            //Do something with your QR code here...
+            DispatchQueue.main.async {
+                self.qrCode.backgroundColor = UIColor(patternImage: qrCodeImage!)
+            }
+        }
+        
+        // Displays QR Code UI View
+        self.qrCode.layer.isHidden = false
+        self.dimScreen.layer.isHidden = false
+
+        // Or display the QR code directly in a share sheet
+        // qrCode.showShareSheetWithQRCode(from: self, anchor: nil, universalObject: buo, linkProperties: lp) { error in
+            // Showing a share sheet with the QR code
+        //}
+    }
+    
+    // Changes background color and sends Branch event
+    @objc func changeBackgroundColor(_ sender: UIButton) {
+        let event = BranchEvent.customEvent(withName: "Custom Change Background Color Event")
+        event.eventDescription = "User clicked the puchase button to send a custom purchase event"
+        event.customData = [
+            "Custom_Event_Property_key1" : "Custom_Event_Property_val1",
+            "App_Type" : "Ios_Sample_App"
+        ]
+        event.logEvent()
+        
+        // logout
+        Branch.getInstance().logout() // or .logoutWithCallback() to customize further
+        
+        self.showToast(message: "Change Background Color Event sent!", font: UIFont(name: "Raleway-Regular", size: 16)!)
+        view.backgroundColor = colors.randomElement()
+    }
+    
+    private let colors: [UIColor] = [
+        .systemRed,
+        .systemCyan,
+        .systemMint,
+        .systemYellow,
+        .systemGreen,
+        .systemPink,
+        .systemOrange,
+        .systemPurple
+    ]
+    
+    // ---------- Setup Constraints and Navigation Bar ----------
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        NSLayoutConstraint.activate([
```

Comment:
For the constraints, I'd recommend trying to stick with leading and trailing constraints instead of width and centerX. Otherwise, it may look wrong on certain device sizes or lead to redundant constraints.

Reply:
Modified all width and centerX constraints to leading & trailing constraints